### PR TITLE
system_state: expand vehicle telemtry interfaces

### PR DESCRIPTION
**Pull request by GuillaumeLaine**
### Changes

- Expand vehicle telemetry interfaces by introducing the following subscription classes:
  - `Battery`
  - `HomePosition`
  - `LandDetected`
  - `VehicleStatus`
  - `VtolStatus`
- Modify rtl_replacement example to use `LandDetected` subscription class
- (Refactor) Move `odometry/subscription.hpp` -> `utils/subscription.hpp` as it is now used in both `odometry/` and `vehicle_state/` modules

### Testing

- rtl_replacement example works as expected in SITL
- Message compatibility OK with new added messages. Checked APX4 (tag: `v1.14.0-3.1.1-beta4`) against px4_msgs (tag: `3.1.0`), both latest.

### CODE REVIEW

**Review by bkueng — COMMENTED** [review]
Nice

File: px4_ros2_cpp/include/px4_ros2/vehicle_state/vehicle_status.hpp [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+   *
+   * @return true if the vehicle is armed, false otherwise
+  */
+  bool armed() const
+  {
+    return last().arming_state == px4_msgs::msg::VehicleStatus::ARMING_STATE_ARMED;
+  }
+
+  /**
+   * @brief Get the vehicle's current active flight mode.
```
[the remark this answers is not in the document]
Comment (GuillaumeLaine) [reply px4_ros2_cpp/include/px4_ros2/vehicle_state/vehicle_status.hpp:39]: Good point :+1:

File: px4_ros2_cpp/include/px4_ros2/vehicle_state/vtol_status.hpp [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+ *
+ * @ingroup vehicle_state
+ */
+class VtolStatus : public Subscription<px4_msgs::msg::VtolVehicleStatus>
+{
+public:
+  explicit VtolStatus(Context & context);
+
+  /**
+   * @brief Check if VTOL is in an undefined state.
```
[the remark this answers is not in the document]
Comment (GuillaumeLaine) [reply px4_ros2_cpp/include/px4_ros2/vehicle_state/vtol_status.hpp:29]: I wasn't aware, that's good to know :bulb:

File: px4_ros2_cpp/src/vehicle_state/battery.cpp [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+#include <px4_ros2/vehicle_state/battery.hpp>
+
+namespace px4_ros2
+{
+
+Battery::Battery(Context & context)
+: Subscription<px4_msgs::msg::BatteryStatus>(context, "fmu/out/battery_status")
+{
+  RequirementFlags requirements{};
+  context.setRequirement(requirements);
```
[the remark this answers is not in the document]
Comment (GuillaumeLaine) [reply px4_ros2_cpp/src/vehicle_state/battery.cpp]: I moved the constructors to the headers given that they are now empty. As a consequence I've also added the include/ dir in the `check-used-topics.py` script, not sure this is desired. If not I'll separate the constructors again

**Review by bkueng — APPROVED** [review]
Thanks, looks all good.
Can you squash the last commit into the previous ones?

**Pull request merged**